blockmanager.go

```diff
@@ -692,8 +699,14 @@ func (b *blockManager) handleBlockMsg(bmsg *blockMsg) {
 	// request more blocks using the header list when the request queue is
 	// getting short.
 	if !isCheckpointBlock {
+		peerRequestedBlocks := 0
+		for _, blockPeer := range b.requestedBlocks {
+			if blockPeer == bmsg.peer { // peer matches
+				peerRequestedBlocks++
+			}
+		}
 		if b.startHeader != nil &&
-			len(bmsg.peer.requestedBlocks) < minInFlightBlocks {
+			peerRequestedBlocks < minInFlightBlocks {
```

Comment:
Same thing here.  When you change the structures to categorize, this can remain a simple len instead of having to do the counting dance above.